src/reconcile/rec_visibility.c

```diff
@@ -100,14 +86,23 @@ __rec_append_orig_value(WT_SESSION_IMPL *session, WT_PAGE *page, WT_INSERT *ins,
      * If we don't have a value cell, it's an insert/append list key/value pair which simply doesn't
      * exist for some reader; place a deleted record at the end of the update list.
      */
-    append = NULL; /* -Wconditional-uninitialized */
-    size = 0;      /* -Wconditional-uninitialized */
+    append = tombstone = NULL; /* -Wconditional-uninitialized */
+    size = 0;                  /* -Wconditional-uninitialized */
     if (unpack == NULL || unpack->type == WT_CELL_DEL)
         WT_RET(__wt_update_alloc(session, NULL, &append, &size, WT_UPDATE_TOMBSTONE));
     else {
         WT_RET(__wt_scr_alloc(session, 0, &tmp));
         WT_ERR(__wt_page_cell_data_ref(session, page, unpack, tmp));
         WT_ERR(__wt_update_alloc(session, tmp, &append, &size, WT_UPDATE_STANDARD));
+        append->start_ts = append->durable_ts = unpack->start_ts;
```

Comment:
I don't think there is a need for a `tombstone` record for every append record. The `tombstone` is required only for history store records to represents the stop time pair of the update. Whereas for the data updates, the next update in the list represents in the stop time pair. The added code can be removed.

Reply:
Added a comment